lib/ide/parser/src/jsclient.rs

```diff
@@ -59,15 +64,23 @@ impl Client {
 
 
 impl IsParser for Client {
-    fn parse(&mut self, program:String, ids:IdMap) -> api::Result<api::Ast> {
+    fn parse(&mut self, program:String, ids:IdMap) -> api::Result<Ast> {
         let ast = || {
             let json_ids = serde_json::to_string(&ids)?;
             let json_ast = parse(program,json_ids)?;
             let      ast = serde_json::from_str(&json_ast)?;
-
             Result::Ok(ast)
         };
-
         Ok(ast()?)
     }
+
+    fn parse_with_metadata
+    (&mut self, program:String) -> api::Result<api::ModuleWithMetadata> {
+        let result = || {
+            let json   = &parse_with_metadata(program)?;
+            let module = serde_json::from_str(&json)?;
+            Result::Ok(module)
+        };
+        Ok(result()?)
```

Comment:
what is this lambda for ?

Reply:
It has the same purpose as in the function about. To convert jsclient::Result to api::Result